llama_stack/distribution/providers.py

```diff
@@ -32,28 +34,34 @@ async def initialize(self) -> None:
 
     async def list_providers(self) -> ListProvidersResponse:
         run_config = self.config.run_config
+        safe_config = StackRunConfig(**redact_sensitive_fields(run_config.model_dump()))
```

Comment:
bodies of list and inspect are now nearly identical. can we move (most of) the code to a helper function and reuse it? the only difference is that inspect matches on the ID.

Reply:
done